### DISCUSSION TITLE
CLI でプロジェクトを作成する際最新のバージョンのテンプレートを作成するようにしてほしい

### DISCUSSION THREAD

**Discussion by koichimurakami** [mbc-net/mbc-cqrs-serverless#33] [Ideas]
現在はCLI をインストールした後にプロジェクトを作成するとCLIのバージョンのプロジェクトが作成される。
パッケージがバージョンアップされていた場合にCLIをバージョンアップせずにプロジェクトを作成すると古いプロジェクトが作成されるためCLIを必ずバージョンアップしてからプロジェクトを作成するのは不便である。
そのため、CLIでプロジェクトを作成する場合最新のバージョンのテンプレートからプロジェクトを作成するようにしてほしい。

**Comment by Tuan-Anh-Nguyen-pionero** [comment]
@koichimurakami 
CLI改善の件については、設計図通りにご提案いたします。
![cli-version](https://github.com/user-attachments/assets/0ffef96e-0f24-4e9c-a26c-36400534b521)

ご確認いただけますと幸いです。
よろしくお願いいたします。

**Reply by koichimurakami** [reply]
ありがとうございます。
こちらでお願いします！

**Reply by baquan1708** [reply]
## Reason: 
- `--version` is a commonly reserved flag for displaying the CLI’s own version. Most CLI tools use `--version` to output the tool’s current version, and users might expect this functionality.

## Solution:
- Choose another flag for setting the framework version.
- Suggested alternatives:
  - `--framework-version` / `-fv`
  - `--target-version` / `-tv`
  - `--use-version` / `-u`

## Question:
- Do we agree on using the new flag for specifying framework version?

**Reply by koichimurakami** [reply]
I agree. `--version` should be the fremework version.
How about `@version` as you know npm packages.

When you  use the specific version, command `mbc new project_name@version`.
Is it possible?

**Reply by baquan1708** [reply]
Yes, it is possible. In this case, the user must at least provide a `project_name` when running `mbc new` command.

So, there are 2 usages:
- `mbc new project_name`: use the latest framework version.
- `mbc new project_name@version`: use this version for project creation.

**Reply by koichimurakami** [reply]
That's good. Could you imprement the logic?

**Reply by baquan1708** [reply]
Yes, I could.

**Reply by baquan1708** [reply]
I've just created a pull request https://github.com/mbc-net/mbc-cqrs-serverless/pull/36 for this feature. Could you review it?

cc: @Tuan-Anh-Nguyen-pionero

**Reply by koichimurakami** [reply]
Thanks.
I've accepted the pull request.

**Reply by koichimurakami** [reply]
Release Info in Japanese: https://www.mbc-net.com/update-mbc-cqrs-serverless-cli-0-1-22-beta/
